server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/anker/AnkerExtension.java

```diff
@@ -0,0 +1,133 @@
+import com.generalbytes.batm.server.extensions.extra.anker.wallets.ankerd.AnkerRPCWallet;
+
+import java.math.BigDecimal;
+import java.util.*;
+
+public class AnkerExtension extends AbstractExtension {
+    private static final CryptoCurrencyDefinition DEFINITION = new AnkerDefinition();
+
+
+    @Override
+    public String getName() {
+        return "BATM Anker extra extension";
+    }
+
+    @Override
+    public IWallet createWallet(String walletLogin) {
+        if (walletLogin != null && !walletLogin.trim().isEmpty()) {
+            StringTokenizer st = new StringTokenizer(walletLogin, ":");
+            String walletType = st.nextToken();
+
+            if ("ankerd".equalsIgnoreCase(walletType)) {
+                String protocol = st.nextToken();
+                String username = st.nextToken();
+                String password = st.nextToken();
+                String hostname = st.nextToken();
+                String port = st.nextToken();
+                String accountName = "";
+                if (st.hasMoreTokens()) {
+                    accountName = st.nextToken();
+                }
+
+
+                if (protocol != null && username != null && password != null && hostname != null && port != null && accountName != null) {
+                    String rpcURL = protocol + "://" + username + ":" + password + "@" + hostname + ":" + port;
+                    return new AnkerRPCWallet(rpcURL, accountName);
+                }
+            } else
+            if ("ankerdemo".equalsIgnoreCase(walletType)) {
+
+                String fiatCurrency = st.nextToken();
+                String walletAddress = "";
+                if (st.hasMoreTokens()) {
+                    walletAddress = st.nextToken();
+                }
+
+                if (fiatCurrency != null && walletAddress != null) {
+                    return new DummyExchangeAndWalletAndSource(fiatCurrency, CryptoCurrency.ANK.getCode(), walletAddress);
+                }
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public ICryptoAddressValidator createAddressValidator(String cryptoCurrency) {
+        if (CryptoCurrency.ANK.getCode().equalsIgnoreCase(cryptoCurrency)) {
+            return new AnkerAddressValidator();
+        }
+        return null;
+    }
+
+    @Override
+    public IRateSource createRateSource(String sourceLogin) {
+        if (sourceLogin != null && !sourceLogin.trim().isEmpty()) {
+            StringTokenizer st = new StringTokenizer(sourceLogin,":");
+            String exchangeType = st.nextToken();
+
+            if ("ankfix".equalsIgnoreCase(exchangeType)) {
+                BigDecimal rate = BigDecimal.ZERO;
+                if (st.hasMoreTokens()) {
+                    try {
+                        rate = new BigDecimal(st.nextToken());
+                    } catch (Throwable e) {
+                    }
+                }
+                String preferedFiatCurrency = FiatCurrency.ZAR.getCode();
```

Comment:
Not sure if calculation from ZAR to other currencies like USD will work. But if ATMs will run in ZAR it will work.